servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HeaderUtils.java

```diff
@@ -81,28 +81,18 @@ static boolean canAddRequestContentLength(final StreamingHttpRequest request) {
 
     static boolean canAddResponseContentLength(final StreamingHttpResponse response,
                                                final HttpRequestMethod requestMethod) {
-        return canAddContentLength(response) && shouldAddZeroContentLength(response.status().code(), requestMethod)
+        return canAddContentLength(response) && responseMayHaveContent(response.status().code(), requestMethod)
                 // HEAD requests should either have the content-length already set (= what GET will return) or
                 // have the header omitted when unknown, but never have any payload anyway so don't try to infer it
                 && !isHeadResponse(requestMethod);
```

Comment:
while you are here, can we inline `isHeadResponse`? this method isn't doing anything special